packages/block-library/src/template-part/edit/index.js

```diff
@@ -17,89 +16,48 @@ import {
 } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 import { chevronUp, chevronDown } from '@wordpress/icons';
-import { serialize } from '@wordpress/blocks';
+
 /**
  * Internal dependencies
  */
-import useTemplatePartPost from './use-template-part-post';
 import TemplatePartNamePanel from './name-panel';
 import TemplatePartInnerBlocks from './inner-blocks';
 import TemplatePartPlaceholder from './placeholder';
 import TemplatePartSelection from './selection';
 
 export default function TemplatePartEdit( {
-	attributes: { postId: _postId, slug, theme, tagName: TagName = 'div' },
+	attributes: { slug, theme, tagName: TagName = 'div' },
 	setAttributes,
 	clientId,
 } ) {
-	const initialSlug = useRef( slug );
-	const initialTheme = useRef( theme );
-	const initialContent = useRef();
-
-	// Resolve the post ID if not set, and load its post.
-	const postId = useTemplatePartPost( _postId, slug, theme );
+	const templatePartId = theme + '|' + slug;
 
 	// Set the postId block attribute if it did not exist,
 	// but wait until the inner blocks have loaded to allow
 	// new edits to trigger this.
-	const { innerBlocks, expectedContent } = useSelect(
+	const { isResolved, innerBlocks } = useSelect(
 		( select ) => {
 			const { getBlocks } = select( 'core/block-editor' );
-			const entityRecord = select( 'core' ).getEntityRecord(
-				'postType',
-				'wp_template_part',
-				postId
-			);
+			const entityRecord =
+				theme && slug
+					? select( 'core' ).getEntityRecord(
+							'postType',
+							'wp_template_part',
+							theme + '|' + slug
```

Comment:
Since we are still using the `theme` attribute from the block markup, I think our conflict with themes in nested directories will still be present after these changes.

That is, on the php side we use the ID as ` wp_get_theme()->get_stylesheet() + '|' + slug ` which iirc would result in something like `some-directory/theme-name|template-part-name`.   While here we are pulling the theme straight from the markup which wouldn't have the directory, and it would try to find `theme-name|template-part-name` and never resolve.

Im wondering how we can go about fixing this.  Brainstorming, could we:
* At some step update the template markup so template-part block `theme` attribute has the correct directory path?
* Always assume a theme supplied template part corresponds to the currently active theme, and find it by that value here instead of the block attribute `theme` value?
* Other ideas?

Reply:
:wave: Hi Addie!

> At some step update the template markup so template-part block `theme` attribute has the correct directory path?

That seems a bit fragile :grimacing: Also, how would we determine the 'correct' directory path? Wouldn't this boil down to the same implicit assumption as your other idea -- that the template-part block is part of the currently active theme?

> Always assume a theme supplied template part corresponds to the currently active theme, and find it by that value here instead of the block attribute `theme` value?

Sounds like a fair assumption to me :thinking: At the very least, it might be solid enough initially, and we might revisit later if it turns out there are cases that this doesn't cover?

Reply:
I'll repeat what I said previously. I don't think we should be changing what the "id" of theme is in WordPress in FSE. And that id includes the parent directory unfortunately or fortunately depending on how you see i. Who tells you that you don't have `theme1` and  `custom/theme1` and that these are two different themes?

Reply:
> Always assume a theme supplied template part corresponds to the currently active theme, and find it by that value here instead of the block attribute theme value?

How do you know that it's a theme supplied template part?

Reply:
I do think there's something to be fixed in FSE framework though. Basically the assumption that theme attribute in the theme files should match the theme folder is wrong. Relying on another "id" that is different from the "theme folder" is wrong too though because no one stops me from doing `custom/theme1` and `custom2/theme1` and make changes to the theme files, I believe these should be considered separate themes.

The most promising one is to "drop" the "theme" attribute from the files and in which case,, they refer to the current theme. the downside is that if we do so, edit a template referring to one of these themes and then switch the theme, we'd have a template from themeA referring to a template part from themeB. Maybe there's a way to hook into the saving behavior of all templates and template parts and say:

 - All saved parts blocks should  explicitly have a "theme" attribute
 - All file templates and templates parts should drop the "theme" attribute and can't refer to a template part from another theme.

Reply:
That said, I think this issue is independent from the current PR.

Reply:
> Wouldn't this boil down to the same implicit assumption as your other idea -- that the template-part block is part of the currently active theme?

Ahh, yes it would. 🙃   So that doesn't make much sense then.

> That said, I think this issue is independent from the current PR.

Yes and no.  Yes, it is already a problem before this PR.  However, this PR makes it hard to distinguish between file based and custom template parts.  Being able to distinguish between customized/uncustomized versions gives us some potential ways to resolve the issue.  They may not be the only solutions, but if we are making changes that limit our options for how we can fix the issue its at least worth pointing it out.

> How do you know that it's a theme supplied template part?

With this PR, Im really not sure anymore.

> I don't think we should be changing what the "id" of theme is in WordPress in FSE.

I don't necessarily disagree with you there.  But we do need to find a way to be consistent between where we are requesting and what is being returned from the request so TPs properly resolve when themes are in nested directories.

Reply:
> * All saved template parts blocks should explicitly have a "theme" attribute
> * All file templates and templates parts should drop the "theme" attribute and can't refer to a template part from another theme.

Yeah that could be a way to distinguish between the two and seems fairly reasonable.  But I think it still has that issue you mentioned:

>  edit a template referring to one of these themes and then switch the theme, we'd have a template from themeA referring to a template part from themeB. 

Would it make sense to make it so that when a template is customized/saved, we ensure the template parts it references are saved as well? 🤔

Reply:
> Yes and no. Yes, it is already a problem before this PR. However, this PR makes it hard to distinguish between file based and custom template parts. 

How so? if anything it makes it easier for me. There an `isCustomized` flag on each template/templart part object (both client side and server-side)

Reply:
> How so? 

From the block's perspective before making the request.  Previously we had block attributes that corresponded to whether or not it referenced an uncustomized or customized TP (the presenece/absence of postId).  Now, we have only removed attributes and we don't know about `isCustomized` until after the request is made and the object is retrieved?  Your note above about redefining when the `theme` block attribute is present would fix that gap though.